src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/NewStyleOneTimeLinkController.kt

```diff
@@ -0,0 +1,50 @@
+package org.vaccineimpact.api.app.controllers
+
+import org.vaccineimpact.api.app.RedirectValidator
+import org.vaccineimpact.api.app.app_start.Controller
+import org.vaccineimpact.api.app.context.ActionContext
+import org.vaccineimpact.api.app.repositories.Repositories
+import org.vaccineimpact.api.app.repositories.TokenRepository
+import org.vaccineimpact.api.models.helpers.OneTimeAction
+import org.vaccineimpact.api.security.WebTokenHelper
+import java.time.Duration
+
+class NewStyleOneTimeLinkController(
+        context: ActionContext,
+        private val repo: TokenRepository,
+        private val tokenHelper: WebTokenHelper,
+        private val redirectValidator: RedirectValidator = RedirectValidator())
+    : Controller(context)
+{
+    constructor(context: ActionContext, repositories: Repositories, tokenHelper: WebTokenHelper)
+            : this(context, repositories.token, tokenHelper)
+
+    val serializer = context.serializer
```

Comment:
I think it makes more sense (esp for a clean testing pattern) to leave the `ActionContext` interface as is (i.e. don't expose this property) and have this controller constructor take an instance of `Serializer`. In any case they'll be the same object, the singleton `MontaguSerializer.instance`. Just keeps dependencies clear.

Reply:
:+1: